src/signals/incident-management/components/CheckboxList/CheckboxList.js

```diff
@@ -182,7 +182,11 @@ const CheckboxList = ({
    */
   const handleIndividualCheck = useCallback(
     ({ target }) => {
-      const { checked: targetIsChecked } = target;
+      const { checked: targetIsChecked, id } = target;
+
+      // Firefox & Safari on Mac do not hold focus for non-input elements when clicked
+      // Force focus so that 'Enter' will submit the current selection
+      document.getElementById(id).focus();
```

Comment:
DOM manipulation is not a good practice. Please use `useRef` here or maybe this can be done through the event object?

Reply:
I didn't use `useRef` to avoid creating 100+ refs on rendering the filter. But good suggestion about the event object, makes a lot more sense to use that directly.